extraction_validator/utilities.py

```diff
@@ -0,0 +1,384 @@
+"""
+Lightweight PreLLM KG/JSON validator
+
+Validates:
+- Schema shape and required fields
+- ID format and uniqueness
+- Referential integrity (edges -> nodes, chains alternate node/edge IDs & all IDs exist)
+- Orphan nodes (degree 0 and unused in chains)
+- Duplicate nodes by (type, normalized name)
+"""
+
+import json
+import sys
+import re
+from typing import Dict, Any, List, Set
+
+ALLOWED_ID_RE = re.compile(r"^[A-Za-z0-9._:-]+$")
+
+
+def _is_str(x):
+    return isinstance(x, str)
+
+
+def _is_dict(x):
+    return isinstance(x, dict)
+
+
+def _is_list(x):
+    return isinstance(x, list)
+
+
+def ok(value: bool) -> str:
+    return "PASS" if value else "FAIL"
+
```

Comment:
Add the `upload_and_create_batch` function

```suggestion
async def upload_and_create_batch(batch_file_name: str) -> str:

    client = AsyncOpenAI(api_key=os.getenv("OPENAI_API_KEY"))

    #Upload the file
    with open(batch_file_name, "rb") as f:
        uploaded_file = await client.files.create(
            file=f,
            purpose="batch"
        )

    # Create the batch
    batch = await client.batches.create(
        input_file_id=uploaded_file.id,
        endpoint="/v1/chat/completions",
        completion_window="24h"
    )

    return batch.id
```